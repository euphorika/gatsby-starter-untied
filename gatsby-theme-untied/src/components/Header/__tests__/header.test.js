import React from 'react'
import renderer from 'react-test-renderer'
import { PureHeader as Header } from '../'

describe('Header', () => {
  it('renders correctly', () => {
    const data = {
      allMenuEntryMain: {
        edges: [
          {
            node: {
              name: 'Menu Entry 1',
              link: '#',
              title: 'Title',
            },
          },
          {
            node: {
              name: 'Menu Entry 2',
              link: '#',
              target: '_blank',
            },
          },
        ],
      },
    }
    const siteTitle = 'Site Title'

    const tree = renderer
      .create(<Header siteTitle={siteTitle} data={data} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
