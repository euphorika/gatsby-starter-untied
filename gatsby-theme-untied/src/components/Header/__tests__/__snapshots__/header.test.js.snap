// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div
  className="css-18penj4"
>
  <header
    className="css-1qn1oi3"
  >
    <h1
      className="css-1y2up5a"
    >
      <Link
        className="css-0"
        to="/"
      >
        <div
          className="css-gajhq5"
        >
          O
        </div>
      </Link>
      <Link
        className="css-0"
        to="/"
      >
        Site Title
      </Link>
    </h1>
    <div>
      <button
        className="css-0"
        id="nav-link"
        onClick={[Function]}
      >
        <svg
          viewBox="0 0 17.91 15.39"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h17.91v1H0zM0 7.2h17.91v1H0zM0 14.39h17.91v1H0z"
            fill="primary"
          />
        </svg>
      </button>
      <div
        className="horizontal-align-center vertical-align-center css-1gl1ukg"
        id="menu"
      >
        <button
          className="css-0"
          id="nav-close"
          onClick={[Function]}
        >
          <svg
            fill="primary"
            viewBox="0 0 13.37 13.37"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M13.37.71L12.66 0 6.68 5.98.71 0 0 .71l5.98 5.97L0 12.66l.71.71 5.97-5.98 5.98 5.98.71-.71-5.98-5.98L13.37.71z"
            />
          </svg>
        </button>
        <nav>
          <ul
            className="css-0"
          >
            <li>
              <Link
                activeClassName="active"
                className="css-0"
                title="Title"
                to="#"
              >
                Menu Entry 1
              </Link>
            </li>
            <li>
              <Link
                activeClassName="active"
                className="css-0"
                target="_blank"
                to="#"
              >
                Menu Entry 2
              </Link>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
</div>
`;
